### ISSUE TITLE
sqlfluff 0.6.8 fails on 'double precision' in CREATE TABLE

### ISSUE THREAD

**Issue by danielbprice** [sqlfluff/sqlfluff#1523]
_If this is a parsing or linting issue, please include a minimal SQL example which reproduces the issue, along with the `sqlfluff parse` output, `sqlfluff lint` output and `sqlfluff fix` output when relevant._

## Expected Behaviour

I'm a new user of sqlfluff but I'm excited to try it on my existing SQL.

I would expect sqlfluff to treat "double precision" declarations as valid SQL.

## Observed Behaviour

Unable to parse section.

## Steps to Reproduce

```shell
$ echo 'CREATE TABLE test ( angle double precision );' | sqlfluff.env/bin/sqlfluff lint -
== [stdin] FAIL
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'CREATE
                       | TABLE test ( angle double precisi...'
All Finished 📜 🎉!
$ echo 'CREATE TABLE test ( angle float );' | sqlfluff.env/bin/sqlfluff lint -
All Finished 📜 🎉!
```

## Dialect

ANSI

## Version

```
$ sqlfluff.env/bin/sqlfluff --version
sqlfluff, version 0.6.8
$ sqlfluff.env/bin/python3 --version
Python 3.8.10
```
## Configuration

I don't have any sqlfluff configuration files.
